### PR TITLE
Expose the widget as an extension and enable local development via an in-app extension slot

### PR DESCRIPTION
**Pull request by manuelroemer**
### Description

The important part of this PR is that the widget is now exposed as an extension. While there are no external extension slots (yet), development is already possible via an in-app ExtensionSlot which is enabled in non-production builds. This allows displaying the widget locally via the `/drugorder/patient/:patientUuid` URL (e.g. `https://openmrs-spa.org/openmrs/spa/drugorder/patient/213da954-87a2-432d-91f6-a3c441851726`).

For demo purposes, some changes have been made to routing etc. I expect this to change again soon, but it requires working out how the widget will be integrated into `patient-chart`. For the moment, I simply wanted to make it work locally without overengineering things.

### CODE REVIEW

File: src/dev-extension-slot.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import ReactDOM from 'react-dom';
+import singleSpaReact from 'single-spa-react';
+import { ExtensionSlotReact } from '@openmrs/esm-extension-manager';
+
+const { bootstrap, mount, unmount } = singleSpaReact({
+  React,
+  ReactDOM,
+  rootComponent: () => {
+    return (
+      <main className="omrs-main-content">
```
Comment (FlorianRappl) [inline src/dev-extension-slot.tsx:11]: Do we need this `main`? Can we just return the extension slot?

Comment (FlorianRappl) [reply src/dev-extension-slot.tsx:11]: Maybe just write:

```jsx
rootComponent: () => <ExtensionSlotReact extensionSlotName="patientChartWidgets" />
```

Comment (manuelroemer) [reply src/dev-extension-slot.tsx:11]: Well, kind of: Without the main (or the CSS class, to be specific) the content is partially hidden behind the title bar. I *could* move the `main` into the extension itself, but that might lead to conflicts with the patient-chart later on.

Comment (manuelroemer) [reply src/dev-extension-slot.tsx:11]: For reference: With the main:
grafik

Without the main:
grafik

Comment (FlorianRappl) [reply src/dev-extension-slot.tsx:11]: OK let's leave the main for the moment. Thanks for the explanation!

**Review by FlorianRappl — APPROVED** [review]
Thanks :beers:!

**Pull request merged**